File: src/main/java/nl/martijnklene/api/application/repository/BlogPostRepository.java
package nl.martijnklene.api.application.repository;

import nl.martijnklene.api.application.entity.BlogPost;

import java.util.Collection;
import java.util.UUID;

public interface BlogPostRepository {
    void save(BlogPost blogPost);
    void delete(BlogPost blogPost);
    BlogPost findOneById(UUID id);
    Collection<BlogPost> findByTitle(String title);
    Collection<BlogPost> findAll();
    Collection<BlogPost> findWithOffset(Integer from, Integer limit);
    BlogPost findLastPublishedBlogPost();
}

File: src/main/kotlin/nl/martijnklene/api/application/repository/BlogPostRepository.kt
package nl.martijnklene.api.application.repository

import nl.martijnklene.api.application.model.BlogPost
import java.util.*

interface BlogPostRepository {
    fun insert(blogPost: BlogPost)
    fun update(blogPost: BlogPost)
    fun deleteById(id: UUID)
    fun findOneById(id: UUID): BlogPost?
    fun findByTitle(title: String): Collection<BlogPost>
    fun findAll(): Collection<BlogPost>
    fun findWithOffset(from: Int, limit: Int): Collection<BlogPost>
    fun findLastPublished(): BlogPost?
}
